### PR TITLE
notify on go-tests failure on main and release branches.

### PR DESCRIPTION
**Pull request by jmurret**
### Description

no slack notifications occur for go-tests on main and release branches when they fail.  related we also would like to move the slow flakey tests to only run on merges to main and release branches, so we need a way to have highly visible notifications of failures so the merging person can be aware and take action.

See https://github.com/hashicorp/consul/pull/19628 for flakey test changes.

### CODE REVIEW

File: .github/workflows/go-tests.yml [hunk 1]
```diff
@@ -516,3 +516,14 @@ jobs:
...
             exit 1
           fi
+      - name: Notify Slack
+        if: ${{ failure() && (github.ref_name == 'main' || startsWith(github.ref_name, 'release/')) }}
+        id: slack
+        uses: slackapi/slack-github-action@e28cf165c92ffef168d23c5c9000cffc8a25e117 # v1.24.0
+        with:
+          payload: |
+            {
+              "message": "One or more nightly integration tests have failed on ${{ github.ref_name }} branch. ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```
Comment (kisunji) [inline .github/workflows/go-tests.yml]: Do we want to modify this message to be more generic?

Comment (jmurret) [reply .github/workflows/go-tests.yml]: ty!  fixed in bb5b6853555d1bf725fd70a1c5450082f58849a3

Comment (zalimeni) [reply .github/workflows/go-tests.yml]: Chatted over Zoom, we're thinking:
* Reuse this script to take advantage of author reporting and formatting
* Remove it from the job-level uses bc that will be incredibly noisy if we ever fix it
* Move to workflow-level in our change, update link and pass in the working webhook URL

Gonna pick up on this for a bit while John is in meetings

**Comment by jmurret** [comment]
closing in favor of https://github.com/hashicorp/consul/pull/19640 since CI got borked.

**Pull request closed without merging**